### PR TITLE
xds: add use_system_root_certs to CertificateValidationContext

### PR DESCRIPTION
**Pull request by markdroth**
Commit Message: xds: add use_system_root_certs to CertificateValidationContext
Additional Description: This allows using system root certs in gRPC.  For details, see https://github.com/grpc/proposal/pull/436.
Risk Level: Low
Testing: N/A
Docs Changes: Included in PR
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

File: api/envoy/extensions/transport_sockets/tls/v3/common.proto [hunk 1]
```diff
@@ -389,6 +389,12 @@ message CertificateValidationContext {
   CertificateProviderPluginInstance ca_certificate_provider_instance = 13
       [(udpa.annotations.field_migrate).oneof_promotion = "ca_cert_source"];
 
+  // Use system root certs for validation.
+  // If true, system root certs are used only if neither of the ``trusted_ca``
+  // or ``ca_certificate_provider_instance`` fields are set.
+  // [#not-implemented-hide:]
+  bool use_system_root_certs = 17;
```
Comment (htuch) [inline api/envoy/extensions/transport_sockets/tls/v3/common.proto:396]: I wonder if this should be a message (empty for now) to future proof against potentially wanting some control over system cert details. For example allow or deny listing certain inbuilt certs.

Comment (markdroth) [reply api/envoy/extensions/transport_sockets/tls/v3/common.proto:396]: I'm not opposed to that if you feel strongly, but I don't think it's really necessary.  The real intent of this option is to tell the client to use built-in logic that knows how to find the system root certs without the control plane having to know the details.  If we wanted the control plane to directly manage this, we would just use certificate providers instead.

Comment (htuch) [reply api/envoy/extensions/transport_sockets/tls/v3/common.proto:396]: It's not just control plane but also bootstrap in some cases. I think just a `SystemRoots` empty message for now. I can see folks wanting to do filtering and other stuff from orchestration logic and so on potentially.

Comment (markdroth) [reply api/envoy/extensions/transport_sockets/tls/v3/common.proto]: The bootstrap use-case is a good point.  Okay, changed it to an empty message.

**Comment by jmarantz** [comment]
@htuch ping

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**